platform/windows/display_server_windows.cpp

```diff
@@ -914,6 +914,31 @@ String DisplayServerWindows::clipboard_get() const {
 				GlobalUnlock(mem);
 			}
 		}
+		
+	} else if (IsClipboardFormatAvailable(CF_HDROP)) {
+		HGLOBAL mem = GetClipboardData(CF_HDROP);
+		if (mem != nullptr) {
+			HDROP hDropInfo = (HDROP)GlobalLock(mem);
+			if (hDropInfo != nullptr) {
+				const int buffsize = 4096;
+				WCHAR buf[buffsize];
+
+				int fcount = DragQueryFileW(hDropInfo, 0xFFFFFFFF, nullptr, 0);
+
+				Vector<String> files;
+
+				for (int i = 0; i < fcount; i++) {
+					DragQueryFileW(hDropInfo, i, buf, buffsize);
```

Comment:
Instead of arbitrary 4K buffer size, it should probably use actual size (Godot support long file names and not limited to MAX_PATH):
```c++
LocalVector<wchar_t> buf;
int size = DragQueryFileW(hDropInfo, i, nullptr, 0);
if (size == 0) {
	continue;
}
buf.resize(size);
DragQueryFileW(hDropInfo, i, (LPTSTR)buf.ptr(), buf.size());
```

Reply:
sorry, this was from copying the surrounding code. I don't know much about how windows itself handles these things